components/DatePicker.vue

```diff
@@ -0,0 +1,74 @@
+<script setup lang="ts">
+import { DatePicker as VCalendarDatePicker } from 'v-calendar';
+import type { DatePickerDate } from 'v-calendar/dist/types/src/use/datePicker';
+import 'v-calendar/dist/style.css';
+
+const today = new Date();
+const minDate = new Date(today).setDate(today.getDate() - 7);
+
+const props = defineProps({
+  modelValue: {
+    type: [Date, Object] as PropType<DatePickerDate | null>,
+    default: null,
+  },
+});
+
+const emit = defineEmits(['update:model-value', 'close']);
+
+const date = computed({
```

Comment:
Very nicely done with getter and setter of `computed`! I love it 😍

Another easy approach would be.. to have a `const date = ref(..)`, pass it to `VCalendarDatePicker`, watch `date`, and emit stuff when `date` changes (which seems hairy). But this approach is elegant 💯